prov/gni/src/gnix_auth_key.c

```diff
@@ -36,13 +36,87 @@
 
 #include "gnix_auth_key.h"
 #include "gnix_hashtable.h"
+#include "gnix_shmem.h"
+#include "gnix_bitmap.h"
 #include "gnix.h"
 
 #define GNIX_AUTH_KEY_HASHSEED 0xdeadbeef
 
+#define GAI_VERSION(major, minor) (((major) << 16) + (minor))
+typedef enum gnix_global_auth_info_version {
+	GNIX_GAI_VERSION_1 = GAI_VERSION(1, 0), /* initial version */
+} gnix_global_auth_info_version_t;
+
+#define MAX_VMDH_TAGS 4
+#define MAX_VMDH_REGS 4096
+#define BITMAP_ARR_SIZE(nbits) \
+	(GNIX_BITMAP_BLOCKS(nbits) * sizeof(gnix_bitmap_block_t))
+#define GNIX_VMDH_PATH "/tmp/gnix_vmdh_info"
```

Comment:
I also don't think its a good idea to hardware a filename that is opened in /tmp.  At least something that honors $TMPDIR environment variable.